### ISSUE TITLE
[FR]: Increase the contrast of color tray icon

### ISSUE THREAD

**Issue by Korb** [martinrotter/rssguard#1999]
### Expected Behavior

Foreground: `#000000`
Background: `#F26F4B`
Contrast ratio: **7.143:1**
1.4.3 Contrast (Minimum) (AA)
✅ Pass for large and regular text
1.4.6 Contrast (Enhanced) (AAA)
✅ Pass for large and regular text

### Actual Behavior

![Image](https://github.com/user-attachments/assets/0a4d34f8-f70d-441b-8a44-2fa99c750953)

Foreground: `#4D1A0C`
Background: `#EF5226`
Contrast ratio: **4.021:1**
1.4.3 Contrast (Minimum) (AA)
❌ Pass for large text only. Fail for regular text
1.4.6 Contrast (Enhanced) (AAA)
❌ Fail for large and regular text

The text may be black to begin with. But due to the small font size and antialiasing, it no longer appears black. In fact, even black text would yield a contrast ratio of **5.905:1**, which is also insufficient.

### Environment

- Microsoft Windows 10.0.26200.7705
- `rssguard-devbuild5-22e7c8f7a-qt6-win10.7z`

### WCAG 2.2 (2024-12-12)

- [1.4.3 Contrast (Minimum)](https://www.w3.org/TR/WCAG22/#contrast-minimum)
- [1.4.6 Contrast (Enhanced)](https://www.w3.org/TR/WCAG22/#contrast-enhanced)
- [1.4.11 Non-text Contrast](https://www.w3.org/TR/WCAG22/#non-text-contrast)

---
Related to #457

**Comment by martinrotter** [comment]
Yes.

https://github.com/martinrotter/rssguard/commit/900e50eb3ce32b8f77a2493c33dffdff0e3d5c15

Can you check latest devbuild?

<img width="243" height="65" alt="Image" src="https://github.com/user-attachments/assets/3e0b1a69-9f20-446f-a809-26c9b79a11bf" />

<img width="103" height="62" alt="Image" src="https://github.com/user-attachments/assets/0f556486-d146-40f9-b2c1-36d547a81caf" />

**Comment by Korb** [comment]
|  | 864 | 11 | 8 |
| - | - | - | - |
| Default | ![Image](https://github.com/user-attachments/assets/c913c3ad-73d0-407b-a134-613e47b4245e) | ![Image](https://github.com/user-attachments/assets/d762a227-acfa-4232-a84d-50e888134511) | ![Image](https://github.com/user-attachments/assets/397a0051-6649-4df4-94a7-d0360d8355f4) |
| Monochrome | ![Image](https://github.com/user-attachments/assets/cd7fc879-ea84-42eb-a805-860b33d0ca78) | … | … |
| Monochrome, colored | ![Image](https://github.com/user-attachments/assets/68dd487c-3335-488a-9aa9-d3faf5f3cfcb) | … | … |

**11, black**:
Foreground: `#3D1409` (unfortunately, anti-aliasing again makes the numbers ~~dark gray~~ brown)
Background: `#EF5226`
Contrast ratio: 4.534:1
1.4.3 Contrast (Minimum) (AA)
✅ Pass for large and regular text
1.4.6 Contrast (Enhanced) (AAA)
❌ Pass for large text only. Fail for regular text

**8, black**:
Foreground: `#020000`
Background: `#EF5226`
Contrast ratio: 5.89:1
1.4.3 Contrast (Minimum) (AA)
✅ Pass for large and regular text
1.4.6 Contrast (Enhanced) (AAA)
❌ Pass for large text only. Fail for regular text

**864, white**:
Foreground: `#C5C5C5`
Background: `#1E1E1E`
Contrast ratio: 9.659:1
1.4.3 Contrast (Minimum) (AA)
✅ Pass for large and regular text
1.4.6 Contrast (Enhanced) (AAA)
✅ Pass for large and regular text

I think displaying more than two symbols on the icon is a bad idea. Even at 100 dpi, they become unreadable. The height of the characters is only five pixels. Perhaps a good solution would be to display the 💯 symbol _(of the desired color)_ on the icon in the Taskbar Notification Area when the number of unread articles is `99+`.

---
`rssguard-devbuild5-d44f54f88-qt6-win10.7z`

> RSS Guard 5.0.0 (built on Windows/AMD64)
> Revision: d44f54f88
> OS: Windows 11 Version 25H2
> Qt: 6.10.2 (compiled against 6.10.2)

**Comment by martinrotter** [comment]
I tweaked the code to show max TWO characters in the tray icon. I consider the issue fixed. Also anti-aliasing should be off now.

**Comment by Korb** [comment]
|  | 9 | 89 | >99 |
| - | - | - | - |
| Default | ![Image](https://github.com/user-attachments/assets/19977e79-137b-4cf2-a337-6be68c212993) | ![Image](https://github.com/user-attachments/assets/85eccad3-c1df-44f1-bfca-91756456fb5c) | ![Image](https://github.com/user-attachments/assets/063b84e9-0905-474b-a5fb-6bb89cdd1fa4) |
| Monochrome | ![Image](https://github.com/user-attachments/assets/c29a3ad2-b3f4-45f5-8c5c-18c8c0ba38eb) | ![Image](https://github.com/user-attachments/assets/32d4708a-c86a-4635-bfd4-706719db8c71) | ![Image](https://github.com/user-attachments/assets/9b82b716-6360-4d3d-90e2-c10f5532bfe2) |
| Monochrome, colored | ![Image](https://github.com/user-attachments/assets/59af7ec6-7555-4d3c-8138-0156052649bb) | ![Image](https://github.com/user-attachments/assets/c67547bb-86a4-4180-8811-41bbe395b31a) | ![Image](https://github.com/user-attachments/assets/8c336d5c-8add-49b4-9abd-b9826cfb06ba) |

**OK**! Thank you!

---
`rssguard-devbuild5-21582f272-qt6-win10.7z`
> RSS Guard 5.0.0 (built on Windows/AMD64)
> Revision: 21582f272
> OS: Windows 11 Version 25H2
> Qt: 6.10.2 (compiled against 6.10.2)

**Comment by ask2018** [comment]
@martinrotter I'll try to ask here before creating new issue. Does this mean the the color of the text of new items in tray has been changed from black to white in version 5? Because I'm not sure how for others, but for me it is basically unreadable now. I cannot read white text on orange background and no I don't want to use monochrome icon. Can I change the text color to black somehow?

Icon on left side is version 4, icon on right side is version 5 and that is almost invisible for my eyes.
![Image](https://github.com/user-attachments/assets/88e559b7-b015-4d3f-9801-d3e6a2a9d845)

**Comment by martinrotter** [comment]
@ask2018 Please, read above comments.

Contrast ration is now much better than it was before - for regular orange tray icon and particularly for monochrome icon. But I guess I could make the regular icon more contrastey.

This is relatively subjective thing - for example for me, the previous black text was actually much less readable.

Activate monochrome icon maybe?

**Comment by ask2018** [comment]
@martinrotter Yes I've read the comments above. I just wanted be sure. This is really very subjective thing and it depeds on persons ability to recognize colors (in Czech we have the word "barvocit" for it). For me it is really basically unreadable now and I don't see the number at all if I not really focus on that space on my monitor and look closely. The monochrome icon is also not a good solution for my needs. I have lot of icons in the tray and the monochrome one is there too hidden or how to say it. The best option would be if you let the user to change the color of the text, so it fits the user needs.

**Comment by martinrotter** [comment]
I just tweaked the icon and made it tad darker to increase the contrast.

**Comment by martinrotter** [comment]
<img width="334" height="286" alt="Image" src="https://github.com/user-attachments/assets/e3db4278-c414-4ce4-802b-e683661b23d7" />

**Comment by ask2018** [comment]
Thank you, but this doesn't help much. From 40 cm I just see solid red space and no text inside , I really need the contrast of the black text if the icon is using red/orange color (and that is how I like it, becuase red/orange is good visible on my grey tray). Please if possible give the user the option to change the color of the text.

**Comment by martinrotter** [comment]
There is separate ticket for that. Might do that. In the meantime, perhaps test out monochrome icon. After the edit, the ratio went to a value around 6.07:1, which is not the greatest, but is much better then previous black text in older versions.

**Comment by ask2018** [comment]
I've tested the monochrome icon and as I have said it is not good solution for me. Then the icon got very similar color to other icons and I need to search for it with the eyes. I guess I'll just use version 4 until this is solved somehow, because this is really problem for me, if I don't see the number of new items and the icon on quick look from distance.

**Comment by ask2018** [comment]
Also that change not showing more than 99 items is also not good imho, it should be increased to 999 or better again let the user to change it. There was absolutely no issue for me to read the number on previous color scheme with 100+ items and it was quite useful to see the exact amount of unread items directly in the tray after Ive been few days away from my PC.

**Comment by ask2018** [comment]
I simply patched the binary as a temporary fix for my needs, becuase I don't want to recompile from source and I really like to test rssguard 5. If somebody got broken eyes like me, you can just nop this function in rssguard.dll and you will have the black text back :) 

```
>rssguard.dll
00000000001F83EC:FF->90
00000000001F83ED:15->90
00000000001F83EE:FE->90
00000000001F83EF:BF->90
00000000001F83F0:BC->90
00000000001F83F1:00->90
```

**Comment by Korb** [comment]
@martinrotter, I'm sure that in @ask2018's case, we're not dealing with color blindness, but with a poor monitor or its settings. I checked the current icon color combination for eight types of color blindness and can confirm that this color combination, while not meeting `WCAG` color contrast requirements for regular text, is still quite legible:

| Type of color blindness | Simulated |
| - | - |
| Green-Weak (Deuteranomaly) | ![Image](https://github.com/user-attachments/assets/83b7d022-475f-40d0-97ee-e64e4ba8ca5e) |
| Green-Blind (Deuteranopia) | ![Image](https://github.com/user-attachments/assets/edc4350f-5d3b-4763-8542-fd9bc509c265) |
| Red-Weak (Protanomaly) | ![Image](https://github.com/user-attachments/assets/194044e5-c201-488c-9740-7a1d816970a4) |
| Red-Blind (Protanopia) | ![Image](https://github.com/user-attachments/assets/622eb84a-218e-4fd4-b383-51ab9efca65a) |
| Blue-Weak (Tritanomaly) | ![Image](https://github.com/user-attachments/assets/f023be0b-115c-49cf-83af-8d989452643c) |
| Blue-Blind (Tritanopia) | ![Image](https://github.com/user-attachments/assets/0e5ec6fa-c3ba-4e48-a328-450b124ed2da) |
| Monochromacy (Achromatopsia) | ![Image](https://github.com/user-attachments/assets/c7d0ae37-9e85-4d91-bb12-8d4689dbae43) |
| Blue Cone Monochromacy | ![Image](https://github.com/user-attachments/assets/56f47a1d-3884-4710-a8bb-ed97fb1086a2) |

However, the color contrast in his white on orange example (`#FAF0ED` on `#D75802`) is `3.547:1`, while the black on orange from his same example (`#0C0801` on `#EF5529`) is `5.711:1`, which is indeed much preferable.

---
Keep in mind that you can evaluate user complaints about color combinations yourself—from the perspective of `WCAG`, as the definitive source of truth.

**Comment by martinrotter** [comment]
I have glasses man. I am have 3x dioptrics value and new colors - white text and new background - is much much more readable for me.

I might add configurable background/foreground tray icon color option in the future. I also enlarged the digits even more to take up almost all available space and also re-enabled three digits display. For me, I consider this "fixed" for now.

**Comment by ask2018** [comment]
@Korb my monitor is little bit older model now, but it is highend EIZO monitor. This is really not about monitor or its settings, but about my problem with color recognition and also about myopia. The first one I have since my childhood. White and red/orange combination is for me like single color from distance. The contrast is not the only aspect you need to consider here and there is no definitive source of truth for this except of each person eyes.

@martinrotter I'm not using glasses, but maybe I should. I just don't want. I've patched both problems directly in the binary for my needs, so I can do the same in next update. That means it is not critical for me, if I have this fix. But please try to think about this as something totally depending on persons eyes. Should be very easy to fix this with custom option. And everybody will be happy.

**Comment by martinrotter** [comment]
@ask2018 Yes, nothing personal here. Like I said, I will maybe add configurable colors in the future. It's completely fair to expect someone might have problems reading some small texts.

**Comment by ask2018** [comment]
@martinrotter Yes all fine. I know I'm here the minority. This is hard to describe if I cannot borrow you my eyes for that, but if there is the black text, then I can easily notice something changed on the icon. It is true, I need to go with my head  closer to the monitor to read the text, specially if there are 3 super small numbers for 100+, but if the text is white there, I cannot notice the change at all. It is like single color for me from distance. But as I have said, I can fix this with the patch mentioned above for my needs quickly, so no rush for the fix from my side.

**Comment by ask2018** [comment]
@martinrotter that darker orange/red tray icon in version 5.0.1 makes it even worse for me. Still the same invisible with white text and less visible with black text than in version 5.0.0.

**Comment by martinrotter** [comment]
Yes, well. It seems your needs are bit specific. I will try to make those colors configurable in one of upcoming releases hopefully.

**Comment by Medrantol** [comment]
> Yes, well. It seems your needs are bit specific. I will try to make those colors configurable in one of upcoming releases hopefully.

That would be great. I saw the redesign with the orange background + white font. It's better than before, but still to hard to see for me. I wish there is an option to turn it like light green (like qrss) or another light color with really high contrast. Thanks for the updates.

**Comment by ask2018** [comment]
Just to make it visible as well here. It has been resolved in version [5.2.0](https://github.com/martinrotter/rssguard/releases/tag/5.2.0). Thank you. It is working as expected.

`Tray icon can now use a custom background color and unread-count text color, with an option to reuse the generated icon as the application icon. (https://github.com/martinrotter/rssguard/issues/1973)`
